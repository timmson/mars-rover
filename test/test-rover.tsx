import React from "react"
import renderer from "react-test-renderer"

import Rover from "../src/rover"

describe("Rover should", () => {

	test("render at (0,0) and have direction to east", () => {
		const component = renderer.create(<Rover position={{x: 0, y: 0, direction: 1}} shift={{x: 0, y: 0}}/>)
		expect(component.toJSON()).toMatchSnapshot()
		component.unmount()
	})

})