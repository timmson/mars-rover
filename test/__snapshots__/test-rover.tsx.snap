// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rover should render at (0,0) and have direction to east 1`] = `
<div
  className="rover"
  style={
    {
      "left": "0px",
      "top": "0px",
      "transform": "rotate(90deg) scaleX(-1)",
    }
  }
>
  <img
    alt="rover"
    src="img/rover.png"
  />
</div>
`;
